chapter01/dxl/src/main/java/com/spring/services/HelloWorld.java

```diff
@@ -0,0 +1,29 @@
+package com.spring.services;
+
+
+public class HelloWorld implements EchoService {
+
+	private String message = "Hello, world!";
+
+	/*
+	 * I initially forgot this, but a bean needs a constructor with no arguments,
+	 * otherwise you get a runtime exception when initialising the object.
+	 */
+	public HelloWorld () {
+		super();
```

Comment:
That super() was unintentional, but I still need an empty constructor to be explicitly defined in order to prevent Spring from complaining. That's due to how beans are defined, and if you remove that constructor you get an exception :cry: unless you change the spring xml file so that both beans use the other constructor passing different strings, and removing the default value for `String message`, as it would be useless (always overwritten).